ArrayDictionarySet/ArrayDictionarySet/ViewController.swift

```diff
@@ -440,6 +462,78 @@ class MySet<V:Comparable>: CustomStringConvertible{
 }
 
 
+class HashDictionaryNode<K,V>{
+    var key:K
+    var value:V
+    var hash:Int
+    
+    init(key:K, value:V, hash:Int){
+        self.key = key
+        self.value = value
+        self.hash = hash
+    }
+}
 
+class MyHashDictionary<K:Hashable,V>: CustomStringConvertible{
+    
+    private var itemsArray = [HashDictionaryNode<K,V>?]()
+    
+    init(){
+        itemsArray = Array.init(repeating: nil, count: 1000)
+    }
+    
+    func add(key:K, value:V){
+        
+        let newItemsHash = getHashCode(key: key)
+        let newNode = HashDictionaryNode(key: key, value: value, hash: newItemsHash)
+        
+        itemsArray[newItemsHash] = newNode
+    }
+    
+    func remove(key:K) throws{
+        let remoweItemsHash = getHashCode(key: key)
+        
+        guard let _ = itemsArray[remoweItemsHash] else{
+            throw MyDictionaryErrors.itemIsNotExist
+        }
+        
+        itemsArray.remove(at: remoweItemsHash)
+    }
+    
+    func getItem(key:K) throws-> V{
+        let neededItemsHash = getHashCode(key: key)
+        
+        guard let neededItem = itemsArray[neededItemsHash] else{
+            throw MyDictionaryErrors.itemIsNotExist
+        }
+        
+        return neededItem.value
+    }
+    
+    func getHashCode(key:K) -> Int{
+        let hashValue = abs(key.hashValue)
+        if itemsArray.count != 0 && hashValue > itemsArray.count{
```

Comment:
this check is useless you can always return "return hashValue % itemsArray.count"